rs/rust_canisters/canister_test/src/canister.rs

```diff
@@ -391,16 +366,18 @@ impl<'a> Runtime {
         num_cycles: Option<u128>,
         specified_id: Option<PrincipalId>,
     ) -> Result<Canister<'a>, String> {
-        let canister_id_record: Result<CanisterIdRecord, String> = match specified_id {
+        let create_canister_result: Result<CreateCanisterResult, String> = match specified_id {
             Some(canister_id) => {
                 self.get_management_canister_with_effective_canister_id(canister_id)
                     .update_(
                         ic00::Method::ProvisionalCreateCanisterWithCycles.to_string(),
                         candid,
-                        (ProvisionalCreateCanisterWithCyclesArgs::new(
-                            num_cycles,
-                            specified_id,
-                        ),),
+                        (ProvisionalCreateCanisterWithCyclesArgs {
+                            amount: num_cycles.map(candid::Nat::from),
+                            settings: None,
+                            specified_id: specified_id.map(|x| x.into()),
```

Comment:
Can you do do `DestinationType::from` instead of `into`? This is a policy we have on our team.

Reply:
Done, also for all other cases.